bin/oneoff/backfill_data/backfill_script_ids_channel_tokens_1.rb

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env ruby
+# Backfill existing ChannelTokens to set script_id if it can easily be inferred because there is only one script associated with the level_id
+
+require_relative '../../../dashboard/config/environment'
+require 'optparse'
+
+# We started writing script_id to the table here: https://github.com/code-dot-org/code-dot-org/pull/39855
+# so we don't need to backfill past that point
+MAX_CHANNEL_TOKEN_ID_FOR_BACKFILL = 303_500_000
+
+# Parse options
+options = {
+  start_id: nil,
+  end_id: nil,
+  dry_run: false,
+}
+
+OptionParser.new do |opts|
+  opts.banner = <<~BANNER
+    Usage: #{File.basename(__FILE__)} [options]
+
+    This script backfills script_id on the Channel Tokens table.
+
+    Options:
+  BANNER
+
+  opts.on('--start-id=1234567',
+    Integer,
+    'Id of first entry to backfill (inclusive).'
+  ) do |start_id|
+    options[:start_id] = start_id
+  end
+
+  opts.on('--end-id=1234567',
+    Integer,
+    'Id of last entry to backfill (inclusive).'
+  ) do |end_id|
+    options[:end_id] = end_id
+  end
+
+  opts.on('--dry-run',
+    'Enables read-only mode where no changes are written to the database'
+  ) do |dry_run|
+    options[:dry_run] = dry_run
+  end
+
+  opts.on('-h', '--help', 'Prints this help message') do
+    puts opts
+    exit
+  end
+end.parse!
+puts "Options: #{options}"
+options.freeze
+
+def update_script_ids(start_id, end_id, is_dry_run)
+  puts "backfilling script_ids..."
+  backfill_count = 0
+  unable_to_backfill = 0
+  levels_missing_backfills = []
+
+  ChannelToken.where(id: start_id..end_id).find_each do |channel_token|
+    next if channel_token.script_id.present?
+
+    level = channel_token.level
+    user_id = user_id_for_storage_id(channel_token.storage_id)
+
+    # if the user has only user_level associated with the level, use the script on that user_level
+    if user_id.present? && user_level_script_id = script_id_by_user_level(user_id, level.id)
```

Comment:
Mostly curious, did you see this assignment idiom used elsewhere?  (This is common in Go, but I'm wondering if it is confusing in Ruby.  It kind of looks like we accidentally used `=` instead of `==`.)

Reply:
I found this pattern in other parts of our codebase and like the concise-ness (is that a word?)